### PR TITLE
🗑️ Deprecate `Color` and `PaymentCardNumber`

### PR DESCRIPTION
**Pull request by Kludex**
## Change Summary

- Deprecate `Color` and `PaymentCardNumber`.
- Delete documentation for both - It will be readded from `pydantic-extra-types` side.

Selected Reviewer: @lig

### CODE REVIEW

**Comment by Kludex** [comment]
please review

**Comment by Kludex** [comment]
Actually... Sorry reviewers. I want to include `pydantic-extra-types` documentation here as well. 🤔

Wait a bit.

**Comment by Kludex** [comment]
please review

File: pydantic/types.py [hunk 1]
```diff
@@ -620,6 +620,7 @@ def __str__(self) -> str:
         return self.value
 
 
+@deprecated('The `PaymentCardNumber` class is deprecated, use `pydantic_extra_types.PaymentCardNumber` instead.')
```
Comment (hramezani) [inline pydantic/types.py:623]: Don't we need to add the commented link like `Color` deprecation warning here?
```suggestion
# TODO: Remove the comment when we add a section about `pydantic-extra-types` on the docs.
@deprecated(
    'The `PaymentCardNumber` class is deprecated, use `pydantic_extra_types.PaymentCardNumber` instead.'
    # 'See https://pydantic-docs.helpmanual.io/usage/types/#pydantic-extra-types'
)
```

**Review by hramezani — APPROVED** [review]

**Review by yezz123 — APPROVED** [review]
🚀

**Pull request merged**